src/sentinel1_reader/sentinel1_reader.py

```diff
@@ -0,0 +1,418 @@
+        orbit_number_offset = 73 if  platform_id == 'S1A' else 202
+        track_number = (orbit_number - orbit_number_offset) % 175 + 1
+
+        center_pts, boundary_pts = get_burst_centers_and_boundaries(tree)
+
+    wavelength = isce3.core.speed_of_light / radar_freq
+    starting_range = slant_range_time * isce3.core.speed_of_light / 2
+    range_pxl_spacing = isce3.core.speed_of_light / (2 * range_sampling_rate)
+
+    # load individual burst
+    burst_list_elements = tree.find('swathTiming/burstList')
+    n_bursts = int(burst_list_elements.attrib['count'])
+    bursts = [[]] * n_bursts
+    sensing_starts = [[]] * n_bursts
+    sensing_times = [[]] * n_bursts
+    for i, burst_list_element in enumerate(burst_list_elements):
+        # TODO make below @classmethod in sentinel1_burst_slc?
+        # get burst timing
+        sensing_start = as_datetime(burst_list_element.find('azimuthTime').text)
+        sensing_starts[i] = sensing_start
+        sensing_times[i] = as_datetime(burst_list_element.find('sensingTime').text)
+        dt = sensing_times[i] - ascending_node_time
+        id_burst = int((dt.seconds + dt.microseconds / 1e6) // burst_interval)
+
+        # choose nearest azimuth FM rate
+        d_seconds = 0.5 * (n_lines - 1) * azimuth_time_interval
+        sensing_mid = sensing_start + datetime.timedelta(seconds=d_seconds)
+        az_fm_rate = get_nearest_polynomial(sensing_mid, az_fm_rate_list)
+
+        # choose nearest doppler
+        poly1d = get_nearest_polynomial(sensing_mid, doppler_list)
+        lut2d = doppler_poly1d_to_lut2d(poly1d, starting_range,
+                                        range_pxl_spacing, (n_lines, n_samples),
+                                        azimuth_time_interval)
+        lut1d = isce3.core.avg_lut2d_to_lut1d(lut2d)
+        doppler = Doppler(poly1d, lut2d)
+
+        # get orbit from state vector list/element tree
+        sensing_duration = datetime.timedelta(
+            seconds=n_samples * azimuth_time_interval)
+        orbit = get_burst_orbit(sensing_start, sensing_start + sensing_duration,
+                                osv_list)
+
+        # determine burst offset and dimensions
+        # TODO move to own method
+        first_valid_samples = [int(val) for val in burst_list_element.find('firstValidSample').text.split()]
+        last_valid_samples = [int(val) for val in burst_list_element.find('lastValidSample').text.split()]
+
+        first_valid_line = [x > 0 for x in first_valid_samples].index(True)
+        n_valid_lines = [x > 0 for x in first_valid_samples].count(True)
+        last_line = first_valid_line + n_valid_lines - 1
+
+        first_valid_sample = max(first_valid_samples[first_valid_line],
+                                 first_valid_samples[last_line])
+        last_sample = min(last_valid_samples[first_valid_line],
+                          last_valid_samples[last_line])
+        n_valid_samples = last_sample - first_valid_sample
```

Comment:
Same comment as above. If `last_sample` is `3` and `first_valid_sample` is `1`, shouldn't `n_valid_samples` equal `3` rather than `2`?